service/build.go

```diff
@@ -24,12 +24,19 @@ import (
 
 // getUiTaskCache takes a build object and returns a slice of
 // uiTask objects suitable for front-end
-func getUiTaskCache(build *build.Build, tasks []task.Task) ([]uiTask, error) {
+func getUiTaskCache(b *build.Build) ([]uiTask, error) {
+	tasks, err := task.FindAll(task.ByBuildId(b.Id))
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get tasks for build")
+	}
+	if tasks == nil {
```

Comment:
nit: can you check len(tasks) == 0 to cover both cases

Reply:
done